models/staging/stg_query_history.sql

```diff
@@ -72,7 +75,8 @@ from {{ source('snowflake_account_usage', 'query_history') }}
 
 {% if is_incremental() %}
     -- must use end time in case query hasn't completed
-    where end_time > (select max(end_time) from {{ this }})
+    -- add lookback window of 2 days to account for late arriving queries
+    where end_time > (select dateadd(day, -2, coalesce(max(end_time), '1970-01-01') ) from {{ this }})
```

Comment:
serverless tasks take some time to arrive in the query history. I'm still working out the exact delay, but starting here.